Migrate to peermap for discovery and check arch in request

Kunde21 opened the pull request:
Peer discovery was noisy and buggy.  Requests weren't being sent to all
peers and some peers weren't being registered.

Integrated peerdiscovery solution and added a ping-back mechanism to register
all peers with the broadcaster, making startup registration much more thorough.

Cross-platform functionality provided by parsing pacman configuration for arch
and checking it against the request.  Minimal implementation that assumes the
standard repository format.  In the future, this may fit better as a header.

Kunde21 committed:
migrate to peermap for discovery and check arch in request

Peer discovery was noisy and buggy.  Requests weren't being sent to all
peers and some peers weren't being registered.

Integrated peerdiscovery solution and added a ping-back mechanism to register
all peers with the broadcaster, making startup registration much more thorough.

Cross-platform functionality provided by parsing pacman configuration for arch
and checking it against the request.  Minimal implementation that assumes the
standard repository format.  In the future, this may fit better as a header.

Kunde21 committed:
refactor to configure in a single location

Minimize the logic in main package and centralize configuration.

Moved configuration and mirrorlist to more standard naming convention.

rakoo commented:
Hey @Kunde21, thanks for being interested in this ! As you have guessed I'm not maintaining it anymore, simply because I don't need it anymore (I only have a single arch machine at home). I can't even test your change. I see you've already taken ownership of the AUR package. Would you like to be the official paclan repo as well ? I can put the redirect in the README, and you can put your own repo in the PKGBUILD.

Kunde21 commented:
Thanks for the work getting this created.  I wasn't paying attention to the dates until I created this PR, but I'm definitely willing to take over the official repo.
I would appreciate the redirect, and I'll make a note in the README in my fork, too.

Kunde21 committed:
bugfix: cache directory not loading correctly

Configured and default cache directory were not being loaded into
config, causing packages not to be discovered.

rakoo commented:
Pushed !

I won't take the trouble of merging your PR, it wouldn't add any value.
